crates/bevy_audio/src/audio.rs

```diff
@@ -267,6 +274,15 @@ impl AudioPlayer<AudioSource> {
     since = "0.15.0",
     note = "Use the `AudioPlayer` component instead. Inserting it will now also insert a `PlaybackSettings` component automatically."
 )]
+// NOTE: When removing this, please remove the `#[expect(deprecated)]` at the top of this file.
+//
+// For whatever reason, this struct counts as a use of itself, causing rustc to lint about a use of
+// a deprecated struct. However, adding an `#[expect(deprecated)]` to this strict causes rustc to
+// say that the lint expectation is unfulfilled.
+//
+// The only solution I could find to this was to mark the whole module as expecting a deprecated
+// lint. So, if this item is ever removed, please remove the `#[expect(deprecated)]` at the top of
+// this file.
```

Comment:
I'm gonna be honest: I have NO idea what's happening with rustc's linter here. It's marking a deprecated item's definition as a use of a deprecated item, *and* it's marking the impls for that deprecated item as a use of that deprecated item.

But I tried my best to silence the lints, while simultaneously changing all `allow`s to `expect`s, and giving them all `reason = "..."`.

So I hope this comment and the module-wide `expect` are okay, because they were the best I could do.